packages/editor/src/components/post-preview-button/index.js

```diff
@@ -190,11 +191,18 @@ export default compose( [
 		const {
 			getPostType,
 		} = select( 'core' );
+
+		let previewLink = getAutosaveAttribute( 'preview_link' );
+		const featuredImageId = getEditedPostAttribute( 'featured_media' );
+		if ( previewLink && featuredImageId ) {
+			previewLink = addQueryArgs( previewLink, { _thumbnail_id: featuredImageId } );
```

Comment:
If this is how Core does it, I'm totally fine with the approach.

Reply:
I'm not certain though how #11409 would fix for non published posts.

Reply:
I can confirm that you can't see the featured image until your draft is saved.